nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/io/ExtendedInfoDeserializer.java

```diff
@@ -45,41 +50,41 @@
     @Override
     public ExtendedInfo deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
 
-        ExtendedInfo info;
         ObjectMapper mapper = (ObjectMapper) jp.getCodec();
         JsonNode node = mapper.readTree(jp);
+        Serializable value;
         switch (node.getNodeType()) {
         case STRING:
-            String value = node.textValue();
+            value = node.textValue();
             try {
-                Date date = Date.from(Instant.parse(value));
-                info = new ExtendedInfoImpl.DateInfo(date);
+                value = Date.from(Instant.parse((String) value));
             } catch (DateTimeParseException e) {
-                info = new ExtendedInfoImpl.StringInfo(value);
+                // ignore
             }
             break;
         case BOOLEAN:
-            info = new ExtendedInfoImpl.BooleanInfo(node.booleanValue());
+            value = node.booleanValue();
             break;
         case NUMBER:
-            Number number = node.numberValue();
-            if (number instanceof Double) {
-                info = new ExtendedInfoImpl.DoubleInfo(node.doubleValue());
-            } else {
-                info = new ExtendedInfoImpl.LongInfo(node.longValue());
+            value = node.numberValue();
+            if (value instanceof Integer) {
+                // convert it to long, it is the original type and json can't differentiate int and long
+                value = Integer.valueOf((Integer) value).longValue();
```

Comment:
`Long.valueOf`

Reply:
I mean, `Long.valueOf(((Integer) value).longvalue())` if you want explicit box/unbox